### ISSUE TITLE
Accepting immutable collections (`Sequence` vs `List` and `Mapping` vs `Dict`)

### ISSUE THREAD

**Issue by kmurphy4** [vemel/mypy_boto3_builder#108]
**Describe your idea**
Hey, I'm not sure if this is a feature request or bug report, but ...

First of all, I just wanted to say that I really appreciate your work on this library -- I've found it extremely useful.

One thing I have noticed is that many (all?) of the API definitions require _mutable_ collections rather than _immutable_ collections.  For example, this doesn't work:

```python
# /tmp/test.py 
from mypy_boto3_ec2.type_defs import EgressOnlyInternetGatewayTypeDef
from mypy_boto3_ec2.type_defs import TagTypeDef
from typing import Sequence

def foo(tags: Sequence[TagTypeDef]) -> EgressOnlyInternetGatewayTypeDef:
    return {
        "Attachments": [],
        "EgressOnlyInternetGatewayId": "bar",
        "Tags": tags,
    }
```

```sh
$ mypy /tmp/test.py 
/tmp/test.py: note: In function "foo":
/tmp/test.py:10:17: error: Incompatible types (expression has type "Sequence[TagTypeDef]", TypedDict item "Tags" has type
"List[TagTypeDef]")  [typeddict-item]
            "Tags": tags,
                    ^
Found 1 error in 1 file (checked 1 source file)
```
Though changing `foo()` to take in a `List[TagTypeDef]` solves the problem.

Is there a specific reason that these functions take in `List[T]` over `Sequence[T]` (and `Dict[K, V]` over `Mapping[K, V]`)?  Note that you can _pass_ a `List[T]` to a function with a `Sequence[T]` parameter, but not the other way around.

**Additional context**
OS details:
```
$ uname -a
Linux bandos 5.3.0-40-generic #32~18.04.1-Ubuntu SMP Mon Feb 3 14:05:59 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.5 LTS
Release:	18.04
Codename:	bionic
```
For this example, I used a fresh virtual environment:
```sh
$ python3.6 -m venv /tmp/venv
$ source /tmp/venv/bin/ac
$ source /tmp/venv/bin/activate
$ pip install mypy==0.910 boto3-stubs[all]==1.18.42
```
Please let me know if there's more information I can provide!

**Comment by vemel** [comment]
Thank you for the report!

It makes perfect sense to me. I will take a look if it is safe to do. Also, I should add `typing_extensions` fallback for `py36/py37`

**Comment by kmurphy4** [comment]
Thanks for the quick reply!

I don't believe you need `typing_extensions`, as [`Sequence`](https://docs.python.org/3.6/library/typing.html#typing.Sequence) and [`Mapping`](https://docs.python.org/3.6/library/typing.html#typing.Mapping) are both in the std `typing` module in 3.6.

**Comment by vemel** [comment]
It took more effort to implement than I expected since return types should still be mutable (`List`/`Dict`), so `Sequence`/`Mapping` should affect only argument types.

But it was actually fun to make this change and should also be quite useful. The changes will be included in today's` boto3-stubs 1.18.43` release.

Thank you for the feature request!

**Comment by kmurphy4** [comment]
Woo!  That's awesome!  Thanks so much :heart:
